url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/Trading_Engine/myDatabase.java

```diff
@@ -0,0 +1,571 @@
+			PreparedStatement tradePre = con.prepareStatement(tradeQuery);
+			con.setAutoCommit(false);
+			int count = 0;
+			int enter = 0;
+			int ask = 0;
+			int bid =0;
+			int amend = 0;
+			int delete = 0;
+			int trade = 0;
+			int other = 0;
+			String checkLastTime = "";
+			while ((st=br.readLine())!=null){
+				String[] insertElement = st.split(",");
+				if(insertElement[3].equalsIgnoreCase("ENTER") ){
+					enter++;
+					if(insertElement[12].equalsIgnoreCase("A")){
+						ask++;
+					}else{
+						bid++;
+					}
+				}else if(insertElement[3].equalsIgnoreCase("AMEND") ){
+					amend++;
+				}else if(insertElement[3].equalsIgnoreCase("DELETE") ){
+					delete++;
+				}else if(insertElement[3].equalsIgnoreCase("TRADE") ){
+					insertQuery(tradePre,insertElement);
+					trade++;
+				}else{
+					other++;
+				}
+				count++;
+				insertQuery(orderBookQuery,insertElement);
+				if(count%1000 == 0){
+					orderBookQuery.executeBatch();
+				}
+				if(trade%1000 == 0){
+					tradePre.executeBatch();
+				}
+				checkLastTime = insertElement[2];
+			}
+			//System.out.println("checkLastTime: " + checkLastTime);
+			orderBookQuery.executeBatch();
+			tradePre.executeBatch();
+			con.commit();
+			String stockName = "default name";
+			Date currentDate = null;
+			Time startTime = null;
+			Time endTime = null;
+			ResultSet firstLine = getResultSet("select * from " + dbName + " order by Entry_Time asc limit 1");
+			if(firstLine.next()){
+				stockName = firstLine.getString(1);
+				currentDate = firstLine.getDate(2);
+				startTime = firstLine.getTime(3);
+			}
+			firstLine.close();
+			
+			overviewResult.add(stockName);
+			overviewResult.add(currentDate.toString());
+			overviewResult.add(startTime.toString() + " to " + checkLastTime.substring(0, 8));
+			overviewResult.add(Integer.toString( count));
+			overviewResult.add(Integer.toString(enter));
+			overviewResult.add(Integer.toString(ask));
+			overviewResult.add(Integer.toString(bid));
+			overviewResult.add(Integer.toString(trade));
+			overviewResult.add(Integer.toString(amend));
+			overviewResult.add(Integer.toString(delete));
+
+			System.out.println("Result of table " + dbName + " :");
+			System.out.println("i1 length: " + count + " == " + (enter + amend + delete + trade + other));
+			System.out.println("enter: " + enter + " ask: " + ask + " bid " + bid);
+			System.out.println("amend: " + amend);
+			System.out.println("delete: " + delete);
+			System.out.println("trade: " + trade);
+
+			orderBookQuery.close();
+		}catch(Exception e){
+			System.out.println("In insertAllDatabase: " + e);
+		}
+	}
+	public LinkedList<String> insertAll(File f, String dbName) {
+		LinkedList<String> overviewResult = new LinkedList<String>();
+		deleteAllTables();
+		try {
+			Statement s = connection.createStatement();
+			BufferedReader br = new BufferedReader(new FileReader(f));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *br*. The resource is referenced at lines 114, 116, 160. The resource is closed at line 167. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).